From 381242275050dbf3405e6d53d0c04af086f22ef0 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:39 -0400
Subject: [fs] nfsd: Remove nfs4_lock_state(): nfsd4_open_downgrade + nfsd4_close

Message-id: <1410898849-22126-156-git-send-email-bfields@redhat.com>
Patchwork-id: 95823
O-Subject: [Patch RHEL7 155/165] nfsd: Remove nfs4_lock_state(): nfsd4_open_downgrade + nfsd4_close
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 36626a2ecfcf23530bf5439679d3a4d040d6cc0d
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 2712cd3..cde72d8 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -4713,7 +4713,6 @@ put_stateid:
  nfs4_put_stid(&stp->st_stid);
 out:
  nfsd4_bump_seqid(cstate, status);
- nfs4_unlock_state();
  return status;
 }
 
@@ -4760,7 +4759,6 @@ nfsd4_open_downgrade(struct svc_rqst *rqstp,
   dprintk("NFSD: %s: od_deleg_want=0x%x ignored\n", __func__,
    od->od_deleg_want);
 
- nfs4_lock_state();
  status = nfs4_preprocess_confirmed_seqid_op(cstate, od->od_seqid,
      &od->od_stateid, &stp, nn);
  if (status)
@@ -4826,7 +4824,6 @@ nfsd4_close(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  dprintk("NFSD: nfsd4_close on file %pd\n", 
    cstate->current_fh.fh_dentry);
 
- nfs4_lock_state();
  status = nfs4_preprocess_seqid_op(cstate, close->cl_seqid,
      &close->cl_stateid,
      NFS4_OPEN_STID|NFS4_CLOSED_STID,
@@ -4842,7 +4839,6 @@ nfsd4_close(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  /* put reference from nfs4_preprocess_seqid_op */
  nfs4_put_stid(&stp->st_stid);
 out:
- nfs4_unlock_state();
  return status;
 }
 
-- 
1.7.1